src/main/java/org/folio/rest/impl/ProtectionHelper.java

```diff
@@ -0,0 +1,142 @@
+package org.folio.rest.impl;
+
+import io.vertx.core.Context;
+import org.folio.HttpStatus;
+import org.folio.orders.rest.exceptions.HttpException;
+import org.folio.orders.utils.ProtectedOperationType;
+import org.folio.rest.jaxrs.model.*;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.stream.Collectors.toList;
+import static org.folio.orders.utils.ErrorCodes.*;
+import static org.folio.orders.utils.HelperUtils.convertIdsToCqlQuery;
+import static org.folio.rest.RestVerticle.OKAPI_USERID_HEADER;
+
+public class ProtectionHelper extends AbstractHelper {
+
+  private AcquisitionsUnitsHelper acquisitionsUnitsHelper;
+  private AcquisitionsUnitAssignmentsHelper acquisitionsUnitAssignmentsHelper;
+  private ProtectedOperationType operation;
+
+
+  public ProtectionHelper(Map<String, String> okapiHeaders, Context ctx, String lang, ProtectedOperationType operation) {
+    super(okapiHeaders, ctx, lang);
+    acquisitionsUnitsHelper = new AcquisitionsUnitsHelper(okapiHeaders, ctx, lang);
+    acquisitionsUnitAssignmentsHelper = new AcquisitionsUnitAssignmentsHelper(okapiHeaders, ctx, lang);
+    this.operation = operation;
+  }
+
+  /**
+   * This method determines status of operation restriction based on ID of {@link CompositePurchaseOrder}.
+   * @param recordId corresponding record ID.
+   *
+   * @return true if operation is restricted, otherwise - false.
+   */
+  public CompletableFuture<Boolean> isOperationRestricted(String recordId) {
+    String userId;
+    if(okapiHeaders != null && (userId = okapiHeaders.get(OKAPI_USERID_HEADER)) != null) {
```

Comment:
In my opinion, it would be reasonable to do this check before calling any API. For example, in `OrdersImpl#postOrdersCompositeOrders`